.github/workflows/typescript-type-generation.yml

```diff
@@ -3,18 +3,76 @@ name: TypeScript Type Generation
 on:
   pull_request:
     paths:
-      - 'openmetadata-spec/src/main/resources/json/schema/**'
+      - "openmetadata-spec/src/main/resources/json/schema/**"
+      - "openmetadata-ui/src/main/resources/ui/src/generated/**"
   workflow_dispatch:
     inputs:
       branch:
-        description: 'Branch to run the workflow on'
+        description: "Branch to run the workflow on"
         required: true
-        default: 'main'
+        default: "main"
         type: string
 
+concurrency:
+  group: typescript-generation-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
 jobs:
   generate-types:
     runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
     steps:
-      - name: Placeholder
-        run: echo "Workflow triggered successfully on branch ${{ github.event.inputs.branch || github.ref_name }}"
\ No newline at end of file
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.inputs.branch || github.event.pull_request.head.ref || github.ref }}
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Switch to branch (if manual trigger)
+        if: github.event_name == 'workflow_dispatch'
+        run: |
+          git checkout ${{ github.event.inputs.branch }}
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: "18"
+
+      - name: Install dependencies
+        run: |
+          yarn install --frozen-lockfile
+
+      - name: Generate TypeScript types
+        run: |
+          cd openmetadata-ui/src/main/resources/ui
+          # Create a symlink to the root node_modules
+          ln -sf ../../../../../node_modules .
+          ./json2ts-generate-all.sh -l true
+
+      - name: Check for changes
+        id: git-check
+        run: |
+          git diff --quiet || echo "changes=true" >> $GITHUB_OUTPUT
+
+      - name: Configure Git
+        if: steps.git-check.outputs.changes == 'true'
+        run: |
+          git config --global user.name 'github-actions[bot]'
+          git config --global user.email 'github-actions[bot]@users.noreply.github.com'
+
+      - name: Commit and push changes
+        if: steps.git-check.outputs.changes == 'true'
+        run: |
+          git add openmetadata-ui/src/main/resources/ui/src/generated/
+          git commit -m "Update generated TypeScript types"
+          git push --force-with-lease
```

Comment:
This will commit to main?

Reply:
No, it'll commit to the branch we have checked out. i.e. will in the PRs